cunumeric/array.py

```diff
@@ -2475,6 +2467,59 @@ def diagonal(
                 raise ValueError("Either axis1/axis2 or axes must be supplied")
         return self._diag_helper(offset=offset, axes=axes, extract=extract)
 
+    @add_boilerplate("indices", "values")
+    def put(
+        self, indices: ndarray, values: ndarray, mode: str = "raise"
+    ) -> None:
+        """
+        Replaces specified elements of the array with given values.
+
+        Refer to :func:`cunumeric.put` for full documentation.
+
+        See Also
+        --------
+        cunumeric.put : equivalent function
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+
+        """
+
+        if values.size == 0 or indices.size == 0 or self.size == 0:
+            return
+
+        if mode not in ("raise", "wrap", "clip"):
+            raise ValueError(
+                "mode must be one of 'clip', 'raise', or 'wrap' "
+                f"(got  {mode})"
+            )
+
+        if mode == "wrap":
+            indices = indices % self.size
+        elif mode == "clip":
+            indices = indices.clip(0, self.size - 1)
+
+        indices = indices._warn_and_convert(np.dtype(np.int64))
+        values = values._warn_and_convert(self.dtype)
+
+        if indices.ndim > 1:
+            indices = indices.ravel()
+
+        if self.shape == ():
+            if values.shape == ():
+                v = values
+            else:
+                v = values[0]
```

Comment:
You need to check here that all values in `indices` are zero; otherwise you would miss an error like:
```
>>> x = np.zeros(())
>>> x
array(0.)
>>> np.put(x, [1], [13])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 180, in put
  File "/Users/mpapadakis/opt/miniconda3/envs/legate/lib/python3.8/site-packages/numpy/core/fromnumeric.py", line 543, in put
    return put(ind, v, mode=mode)
IndexError: index 1 is out of bounds for axis 0 with size 1
```

Reply:
makes sense, thank you